prompts/n2c2_smoking.py

```diff
@@ -0,0 +1,188 @@
+"""
+n2c2 Smoking Status Identification Challenge
+Özlem Uzuner, PhD, Ira Goldstein, MBA, Yuan Luo, MS, Isaac Kohane, MD, PhD, Identifying Patient Smoking Status from Medical Discharge Records, Journal of the American Medical Informatics Association, Volume 15, Issue 1, January 2008, Pages 14–24, https://doi.org/10.1197/jamia.M2408
+Usage:
+python n2c2_smoking.py --indir data_dir/ --outdir output_dir/
+"""
+import os
+import argparse
+from pathlib import Path
+from loguru import logger
+from functools import partial
+import xml.etree.ElementTree as et
+from prompts import DatasetPrompts
+
+_HOMEPAGE = "https://portal.dbmi.hms.harvard.edu/projects/n2c2-nlp/"
+_TRAINING_FILE = "smokers_surrogate_train_all_version2.xml"
+_TEST_FILE = "smokers_surrogate_test_all_groundtruth_version2.xml"
+
+def load_smoking_corpus(path):
+    """
+    Load the data split.
+    :param path: train/test data path
+    """
+    logger.info(f"Loading n2c2 Smoking Status data from {path}")
+    with open(path) as raw:
+        file = raw.read().strip()
+    root = et.fromstring(file)
+    ids = []
+    notes = []
+    labels = []
+    documents = root.findall("./RECORD")
+    for document in documents:
+        ids.append(document.attrib['ID'])
+        notes.append(document.findall('./TEXT')[0].text)
+        labels.append(document.findall('./SMOKING')[0].attrib['STATUS'].lower())
+
+    return [(id,note,label) for id, note, label in zip(ids,notes,labels)]
+    
+
+class SmokingCorpusPrompts(DatasetPrompts):
+    """
+    Create prompts from n2c2 smoking status corpus
+    """
+    def __init__(self, data_root):
+        """Instantiate SmokingCorpusPrompts.
+        :param data_root: Root folder containing target dataset
+        """
+        self.data_root = Path(data_root).resolve()
+        self._init_dataset()
+        self._init_prompts()
+
+    def _init_dataset(self):
+        """
+        Prepare smoking status data.
+        Download data manually from https://portal.dbmi.hms.harvard.edu/projects/n2c2-nlp/", which requires requires user registration and DUA.
```

Comment:
Yeah, I don't know how to get around the manual download aspect here for n2c2 datasets. I think the most generic solution is just having a `fpath` option in the dataset init where a user can point to a downloaded file.